subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/ExtraExecutionGraphDependenciesResolver.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.artifacts.transform;
+
+import org.gradle.api.internal.attributes.ImmutableAttributes;
+import org.gradle.execution.plan.Node;
+import org.gradle.execution.plan.TaskDependencyResolver;
+
+import java.util.Collections;
+import java.util.Set;
+
+public interface ExtraExecutionGraphDependenciesResolver {
```

Comment:
What does this name mean? Can you maybe add Javadoc?

Reply:
Added